Sources/ViewInspector/ViewHosting.swift

```diff
@@ -429,3 +444,16 @@ public extension ViewHosting {
     }
 }
 #endif
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+internal extension MainActor {
+    static func syncRun<T>(action: @MainActor @Sendable () throws -> T) rethrows -> T {
+        if Thread.isMainThread {
+            return try assumeIsolated(action)
+        } else {
+            return try DispatchQueue.main.sync {
```

Comment:
Using `sync` on the main queue is usually a great way to cause a deadlock. IMO we'd be better off just using assumeIsolated without the check and `else` branch. `assumeIsolated` should already give good diagnostics if it's not run on the main thread to help somebody trace through. This could hide those diagnostics behind weird deadlocks -- especially in XCTest which tends to make some interesting choices about what's run on the main thread.

Reply:
Good point, I'll change it